Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py

```diff
@@ -248,12 +271,23 @@ def fetch_mails(client: IMAPClient,
         email_message_object = Email(message_bytes, include_raw_body, save_file, mail_id)
         if not first_fetch_time or email_message_object.date > first_fetch_time:
             mails_fetched.append(email_message_object)
-    return mails_fetched, messages
+            messages_fetched.append(email_message_object.id)
+        else:
+            demisto.debug(
+                f'Skipping {mail_id}. Email date: {email_message_object.date}, criterion SINCE time: {first_fetch_time}'
+            )
+
+    last_message_in_current_batch = uid_to_fetch_from
+    if messages:
+        last_message_in_current_batch = messages[-1]
```

Comment:
I think this should be set also in cases that all messages found have a date less than first_fetch_time

Reply:
it will be set in that case as well
`messages` is a list of all the UIDs that were fetched, and is not filtered by dates

Reply:
Ok. Wasn't clear on the code. Maybe better to call this variable `message_uids` instead of `messages`.